app/qml/map/MapWrapper.qml

```diff
@@ -166,30 +169,41 @@ Item {
     // highlights may end up with dangling pointers to map layers and cause crashes)
     _highlightIdentified.featureLayerPair = null
     _digitizingHighlight.featureLayerPair = null
+    _navigationHighlight.featureLayerPair = null
   }
 
   states: [
     State {
       name: "view"
       PropertyChanges { target: root; isInRecordState: false }
+      PropertyChanges { target: root; isInNavigationState: false }
     },
     State {
       name: "recordFeature"
       PropertyChanges { target: root; isInRecordState: true }
+      PropertyChanges { target: root; isInNavigationState: false }
     },
     State {
       // recording feature in specific layer without option to change the digitized layer.
       // can be used to create linked features in relations, value relations and browse data
       name: "recordInLayerFeature"
       PropertyChanges { target: root; isInRecordState: true }
+      PropertyChanges { target: root; isInNavigationState: false }
     },
     State {
       name: "editGeometry" // of existing feature
       PropertyChanges { target: root; isInRecordState: true }
+      PropertyChanges { target: root; isInNavigationState: false }
```

Comment:
Do we need this `isInNavigationState` property? I think checking `if (state === "navigation")` is sufficient. Property `isInRecordState` is there because multiple states have common behaviour of "recording" and checking against all these states would be exhausting 🙂